CONTRIBUTING.md

```diff
@@ -0,0 +1,165 @@
+- [`flake8`](https://flake8.pycqa.org/en/latest/) for code linting
+- [`mypy`](https://mypy.readthedocs.io/en/stable/) for static type checking
+- [Github Actions](https://github.com/features/actions)
+
+A deep understanding of these tools in not required to effectively contribute to `dbt`, but we recommend checking out the attached documentation if you're interested in learning more about them.
+
+#### virtual environments
+
+We strongly recommend using virtual environments when developing code in `dbt`. We recommend creating this virtualenv
+in the root of the `dbt` repository. To create a new virtualenv, run:
+```sh
+python3 -m venv env
+source env/bin/activate
+```
+
+This will create and activate a new Python virtual environment.
+
+#### docker and docker-compose
+
+Docker and docker-compose are both used in testing. Specific instructions for you OS can be found [here](https://docs.docker.com/get-docker/).
+
+## Running `dbt-bigquery` in development
+
+### Installation
+
+First make sure that you set up your `virtualenv` as described in [Setting up an environment](#setting-up-an-environment).  Also ensure you have the latest version of pip installed with `pip install --upgrade pip`. Next, install `dbt-biguery` latest dependencies:
+
+```sh
+pip install -r dev-requirements.txt
+```
+
+When `dbt` is installed this way, any changes you make to the `dbt` source code will be reflected immediately in your next `dbt` run.
+
+## Testing
+
+### Test commands
+There are a few mehtods for running tests locally.
+
+####`tox`
+`tox` takes care of managing virtualenvs and install dependencies in order to run tests. You can also run tests in parallel, for example you can run unit tests for Python 3.6,Python 3.7, Python 3.8, `flake8` checks and `mypy` checks in parallel with `tox -p`. Also you  can run unit tests for specific python versions with `tox -e py36`. The configuration of theses tests are located in `tox.ini`.
+
+####`pyteest`
+Finally, you can also run  a specifiic test or group of tests using `pytest` directlly. With virtualenv activa and dev dependencies installed you can do things like:
+```sh
+# run specific postgres integration tests
+python -m pytest -m profile_bigquery test/integration/001_simple_copy_test
+# run all unit tests in a file
+python -m pytest test/unit/test_bigquery_adapter.py
+# run a specific unit test
+python -m pytest test/unit/test_bigquery_adapter.py::TestBigQueryAdapter::test_copy_table_materialization_table
+```
+## Updating Docs
+
+Many changes will require and update to the `dbt` docs here [docs](https://github.comdbt-labs/docs.getdbt.com).
+
+
+## Submitting a Pull Request
+
+dbt Labs provides a CI environment to test changes to specific adapters, and periodic maintenance checks of `dbt-core` through Github Actions. For example, if you submit a pull request to the `dbt-redshift` repo, GitHub will trigger automated code checks and tests against Redshift.
```

Comment:
We can be less vague here, since we know their PR is in the `dbt-bigquery` repo and wants to run BigQuery integration tests.

That said, we should say things like:
- If you're a first-time contributor, code checks + unit tests require a maintainer to approve
- A maintainer will need to add the `ok to test` label, at which point your subsequent commits will trigger integration tests in BigQuery

I believe those are true across dbt-core + all adapter plugins. We can decide if it's worth reiterating here, for folks who skim the core contributing guide